src/wolfkrow/core/engine/resolver.py

```diff
@@ -46,26 +169,27 @@ def resolve(self, value):
 
         Args:
             value (str, dict or list): An instance to do replacements for.
+            replacements (dict): The replacements to use for replacing instead of the replacements assigned on this object.
         """
-        # FIXME(TM): Should this deep copy? We're currently changing lists +
-        # dicts in place.
-        replaced_value = value
+        # Not a deep copy, but this is fine because were only ever assigning new values 
+        # to the dictionary, and not modifying existing ones.
+        replaced_value = copy.copy(value)
```

Comment:
I'm trying to "optimize" here be avoiding the more expensive deep copy method... But really the difference will be negligible because we are doing lots of file/network IO operations which are many orders of magnitude slower than this. 

Lets just change this to be deepcopy.